File: typescript/src/problems/s557_reverse_words_in_a_string_iii.ts
/**
 * @link https://leetcode.com/problems/reverse-words-in-a-string-iii/
 * @param s
 */
export function reverseWords(s: string): string {
  const words = s.split(' ').map((word) => {
    return word.split('').reverse().join('');
  });
  return words.join(' ');
}

File: rust/src/problems/s557_reverse_words_in_a_string_iii.rs
struct Solution;

impl Solution {
    pub fn reverse_words(s: String) -> String {
        s.split_whitespace()
            .into_iter()
            .map(|word| word.chars().rev().collect::<String>())
            .collect::<Vec<String>>()
            .join(" ")
    }
}

struct Example {
    input: String,
    output: String,
}

#[test]
pub fn test() {
    let examples = vec![
        Example {
            input: "Let's take LeetCode contest".to_string(),
            output: "s'teL ekat edoCteeL tsetnoc".to_string(),
        },
        Example {
            input: "God Ding".to_string(),
            output: "doG gniD".to_string(),
        },
    ];
    for example in examples {
        assert_eq!(Solution::reverse_words(example.input), example.output);
    }
}
